src/state/index.ts

```diff
@@ -52,26 +57,73 @@ const persistConfig = {
   migrate: createMigrate(migrations, { debug: false }),
 }
 
-const persistedReducer = persistReducer(
-  persistConfig,
-  combineReducers({
-    farms: farmsReducer,
-    pools: poolsReducer,
-    predictions: predictionsReducer,
-    lottery: lotteryReducer,
-    info: infoReducer,
-    nftMarket: nftMarketReducer,
-
-    // Exchange
-    user,
-    transactions,
-    swap,
-    mint,
-    burn,
-    multicall,
-    lists,
-  }),
-)
+const allReducers = combineReducers({
+  farms: farmsReducer,
+  pools: poolsReducer,
+  predictions: predictionsReducer,
+  lottery: lotteryReducer,
+  info: infoReducer,
+  nftMarket: nftMarketReducer,
+
+  // Exchange
+  user,
+  transactions,
+  swap,
+  mint,
+  burn,
+  multicall,
+  lists,
+})
+
+const rootReducer = (state, action) => {
+  if (action.type === 'global/resetUserState') {
+    // eslint-disable-next-line no-param-reassign
+    return {
+      ...state,
+      farms: {
+        ...state.farms,
+        data: state.farms.data.map((farm) => {
+          return {
+            ...farm,
+            userData: {
+              allowance: '0',
```

Comment:
Hmm. When I looked at this code, I realized my idea is bad. Exposing many detail of sub-reducers in the root reducer is quite hard to maintain in the future. For example, if we add more state in the store, we have to maintain this code as well. 

I will think another cleaner solution. 😅

Reply:
We can revert back to the run subreducers after action logic

Reply:
ah with that we still have to assign type :(

Reply:
I think if we let reducers run by assigning state only instead of return, we don't have to assign a type

Reply:
I think we could listen the reset action in each relevant sub-reducer to make it more precise. 

I think this way would fit better in redux paradigm. 

What do you think ? @memoyil

Reply:
Done